### PR TITLE
Add partnercourse related commands

### PR DESCRIPTION
**Pull request by dlathyun**
Add partnercourse related commands and catalogue
TODO: edit SeplendidLogicManager

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/PartnerCourseAddCommandParser.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        return new PartnerCourseAddCommand(partnerCourse);
+    }
+
+    /**
+     * Returns true if all arguments are enclosed in square brackets, and are non-empty.
+     *
+     * @param args Arguments in the format of {@code [args1] [args2] ...}.
+     * @return true if in correct format.
+     */
+    private static boolean areValuesEnclosedAndNonEmpty(String args) {
```
Comment (aliciamichellew) [inline src/main/java/seedu/address/logic/parser/PartnerCourseAddCommandParser.java:52]: I think you can remove this. Cheng Hou puts this function in parserutil

File: src/main/java/seedu/address/logic/parser/PartnerCourseAddCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        }
+
+        return bracketCount == 0;
+    }
+
+    /**
+     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
+     * {@code SeplendidArgumentMap}.
+     */
+    private static boolean areArgumentsPresent(SeplendidArgumentMap argumentMap, SeplendidParameter... parameters) {
```
Comment (aliciamichellew) [inline src/main/java/seedu/address/logic/parser/PartnerCourseAddCommandParser.java:81]: Same for this function

Comment (dlathyun) [reply src/main/java/seedu/address/logic/parser/PartnerCourseAddCommandParser.java]: ok!

**Review by aliciamichellew — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**